File: tools/inspect_oxford.mjs
import { readFile } from "node:fs/promises";

async function importSourceModule(relativePath) {
  const source = await readFile(new URL(relativePath, import.meta.url), "utf8");
  const url = `data:text/javascript;base64,${Buffer.from(source).toString("base64")}`;
  const mod = await import(url);
  return mod.oxfordBoxes || mod.default || [];
}

const oxfordBoxes = await importSourceModule("../src/data/oxfordBoxes.js");
console.log(`Total Oxford boxes: ${oxfordBoxes.length}`);

let fullyGreek = 0;
let englishOrMixed = 0;
const listToTranslate = [];

for (let i = 0; i < oxfordBoxes.length; i++) {
  const box = oxfordBoxes[i];
  const str = JSON.stringify(box);
  const greekLetters = (str.match(/[\u0370-\u03ff\u1f00-\u1fff]/g) || []).length;
  const englishLetters = (str.match(/[a-zA-Z]/g) || []).length;
  
  // Calculate English ratio
  const ratio = englishLetters / (greekLetters + englishLetters + 1);
  if (ratio > 0.3) {
    englishOrMixed++;
    listToTranslate.push({
      index: i,
      id: box.id,
      chapter: box.chapter,
      boxNumber: box.boxNumber,
      title: box.title,
      ratio: ratio.toFixed(2),
      sample: str.slice(0, 100)
    });
  } else {
    fullyGreek++;
  }
}

console.log(`Fully Greek boxes: ${fullyGreek}`);
console.log(`English or Mixed boxes: ${englishOrMixed}`);

console.log("\nBoxes to translate/review:");
for (let i = 0; i < listToTranslate.length; i++) {
  const item = listToTranslate[i];
  console.log(`${i+1}. [Index ${item.index}] ID: ${item.id} | Ch ${item.chapter} Box ${item.boxNumber} | "${item.title}" | Eng Ratio: ${item.ratio}`);
}

File: tools/inspect_oxford.py
import json
import re
import sys

if sys.stdout and hasattr(sys.stdout, "reconfigure"):
    sys.stdout.reconfigure(encoding="utf-8")

with open("src/data/oxfordBoxes.js", "r", encoding="utf-8") as f:
    text = f.read()

print("Length of file (characters):", len(text))
print("First 400 characters:\n", text[:400])

# Match oxfordBoxes
m = re.search(r'export const oxfordBoxes\s*=\s*(\[[\s\S]*?\]);', text)
if m:
    boxes = json.loads(m.group(1))
    print(f"Total boxes: {len(boxes)}")
    for i, b in enumerate(boxes):
        print(f"{i+1}. id={b.get('id')}, title={b.get('title')}, topic={b.get('topic')}")
else:
    print("Could not parse as JSON array directly, inspecting structure...")
